modules/document_repository/ajax/documentEditUpload.php

```diff
@@ -120,8 +114,10 @@
         $comments   = $_POST['commentsEdit'];
         $version    = $_POST['versionEdit'];
 
-        if (empty($category) && $category !== '0') {
-            header("HTTP/1.1 400 Bad Request");
+        // $category is a string representation of a number.
+        // only proceed if its int value is greater than or equal to 0
+        if (intval($category) < 0) {
```

Comment:
still not sure how checking that it is less than 0 is the same as checking its empty and not 0

Reply:
I don't think it's equivalent for strings with an int value less than 0

Reply:
@zaliqarosli I feel like this is better written this way. What we care about is the integer value being at least 0.

Reply:
@johnsaigle could you explain why we care about the integer value being atleast zero?

Reply:
@zaliqarosli Just the way the module was coded. `File_category` in the DB is a value of 0 or higher that effectively represents a category ID. When uploading a file and choosing its category, what gets passed to the back-end isn't the name of the category but its ID. Do some poking around `document_repository` and `document_repository_categories` and you'll see what I mean.

Since $category should only be positive integers, it makes sense to validate this way.

Reply:
i think i understand what's going on now after looking into what $category actually is and how it's stored in the database. from what i see, category should be > 0 (`  `File_category` int(3) ~~unsigned~~ DEFAULT NULL ` ` for `document_repository` table)

Reply:
0 is a valid unsigned variable though, no?

Reply:
if category is always unsigned then my concern above is not valid 👍

Reply:
@johnsaigle yes it is valid, hence why i agree :)